portfolio/src/main/webapp/script.js

```diff
@@ -139,6 +141,29 @@ function formatDate(epochDate) {
 
 }
 
+/**
+ * Deletes all guestbook comments from the website if the 
+ * correct password is entered.
+ */
+function deleteData() {
+  const enteredPassword = document.getElementById('pwd').value;
+  const requestURL = '/delete-data?password=' + enteredPassword;
+  const request = new Request(requestURL, {method: 'POST'});
+  
+  fetch(request).then((response) => {
+    if (response.status === 403) {
```

Comment:
This piece of code assumes that if the status code is not equal to 403 it is a success. 

Though you know from the server code the responses are either 403 or 200 it is not right to make that assumption on the client side. 

Two ways of fixing this

1. Use the positive case for the if condition.

if (response.status == 200) {
  // Success
} else {
 // All failure conditions
}


2. Have a generic error fallback. This allows for a accurate error message for 403 and at the same time for future errors

if (response.status == 200) {
  // Generic success
} else if( response.status == 403) {
 // Forbidden. Indicates password incorrect.
//  Add clear error message to the user
}  else {
 // Unexpected error. Add a generic failure message.
}